### PR TITLE
fix: prevent relocating urls that start with com like /computeMetadata/

### LINKED ISSUE #503: Shade relocation changes the address of metadata server then compute engine credentials get stuck
I have encountered a corner case which emerges when I generate my shaded uber jar. I have realized that shaded jar is not able to get service account credentials in a VM/GKE pod. When I decompiled the shaded jars, I have realized that the gcp metadata server path is prefixed with `/com/google/cloud/spanner/jdbc/shaded` . 
As a solution I have changed pattern `com` to `com.` and I've applied similarly to `io.`, `org.` and `android.`

#### Environment details

1. spanner-jdbc-driver
2. macos
3. Java version: 8
4. spanner-jdbc version(s): master

#### Steps to reproduce

  1. mvn package -Pshade -DskipTests
  2. cd target
  3. unzip google-cloud-spanner-jdbc-*-single-jar-with-dependencies.jar
  4. cd com/google/cloud/spanner/jdbc/shaded/com/google/auth/oauth2
  5. decompile ComputeEngineCredentials.class , you will see 
 ```java
public static String getServiceAccountsUrl() {
    return getMetadataServerUrl(DefaultCredentialsProvider.DEFAULT) + "/com/google/cloud/spanner/jdbc/shaded/computeMetadata/v1/instance/service-accounts/?recursive=true";
  }
```
As you see above the metadata server path is prefixed by shaded address. 

#### Solution
I have excluded the computeMetadata in pom.xml by changing the patterns.
```xml
<relocations>
                <relocation>
                  <pattern>com.</pattern>
                  <shadedPattern>com.google.cloud.spanner.jdbc.shaded.com.</shadedPattern>
                  <excludes>
                    <exclude>com.google.cloud.spanner.**</exclude>
                  </excludes>
                </relocation>
                <relocation>
                  <pattern>android.</pattern>
                  <shadedPattern>com.google.cloud.spanner.jdbc.shaded.android.</shadedPattern>
                </relocation>
                <relocation>
                  <pattern>io.</pattern>
                  <shadedPattern>com.google.cloud.spanner.jdbc.shaded.io.</shadedPattern>
                  <excludes>
                    <exclude>io.grpc.netty.**</exclude>
                  </excludes>
                </relocation>
                <relocation>
                  <pattern>org.</pattern>
                  <shadedPattern>com.google.cloud.spanner.jdbc.shaded.org.</shadedPattern>
                  <excludes>
                    <exclude>org.conscrypt.**</exclude>
                  </excludes>
                </relocation>
              </relocations>
```
#### Any additional information below
I have encountered this while trying to integrate cloud spanner to an old codebase.
Also I have realized that many libraries are dragged by shaded pom so when I add shaded jar as a dependency I have excluded everything:

```xml
<dependency>
      <groupId>com.google.cloud</groupId>
      <artifactId>google-cloud-spanner-jdbc</artifactId>
      <version>2.2.1-shaded</version>
      <classifier>single-jar-with-dependencies</classifier>
        <exclusions>
            <exclusion>
                <groupId>*</groupId>
                <artifactId>*</artifactId>
            </exclusion>
        </exclusions>
    </dependency>
```

### PR DESCRIPTION
**Pull request by yunus**
Fixes #503 

In addition to original problem I have added an exception for all the Netty libraries since I have got errors on service provider which claims that netty libraries are not a subtype.

### CODE REVIEW

**Comment by olavloite** [comment]
@yunus Could you elaborate a little on what exactly the problem is with the netty shading?

Also, for the conventionalcommits check to pass, you need to alter your commit message to include a prefix that indicates the type of change (in this case `fix: ...`)

File: pom.xml [hunk 1]
```diff
@@ -333,26 +333,26 @@
...
+                  <pattern>android.</pattern>
+                  <shadedPattern>com.google.cloud.spanner.jdbc.shaded.android.</shadedPattern>
                 </relocation>
                 <relocation>
-                  <pattern>io</pattern>
-                  <shadedPattern>com.google.cloud.spanner.jdbc.shaded.io</shadedPattern>
+                  <pattern>io.</pattern>
+                  <shadedPattern>com.google.cloud.spanner.jdbc.shaded.io.</shadedPattern>
                   <excludes>
-                    <exclude>io.grpc.netty.shaded.**</exclude>
```
Comment (olavloite) [inline pom.xml:350]: Would you mind elaborating a little bit on why this change is needed? The PR description mentions 'errors on service provider...'. Could you add the specific error message that you are getting?

Comment (yunus) [reply pom.xml:350]: I couldn't replicate the error. so I've rolled back netty related fix. If I encounter again, I open a separate ticket. Sorry for the mess.

Comment (olavloite) [reply pom.xml:350]: No worries. Thanks for the detailed analysis and fix for the shading problem.

**Review by olavloite — APPROVED** [review]
